h/accounts/__init__.py

```diff
@@ -34,8 +35,25 @@ def make_staff(username):
         raise NoSuchUserError
 
 
+def authenticated_user(request):
+    """Return the authenticated user or None.
+
+    :rtype: h.accounts.models.User or None
+
+    """
+    if not request.authenticated_userid:
+        return None
+
+    # pylint: disable=unpacking-non-sequence
```

Comment:
Done, I changed it to raise an exception: https://github.com/hypothesis/h/commit/a2025ccac881ed4e8990c9e5ffa1a8241445ea4f